src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -133,13 +136,25 @@ void OTARequestor::OnQueryImageResponse(void * context, const QueryImageResponse
             return;
         }
 
-        MutableByteSpan updateToken(requestorCore->mUpdateTokenBuffer);
-        CopySpanToMutableSpan(update.updateToken, updateToken);
-        requestorCore->mUpdateVersion = update.softwareVersion;
-        requestorCore->mUpdateToken   = updateToken;
+        if (update.softwareVersion > currentSoftwareVersion)
+        {
+            ChipLogDetail(SoftwareUpdate, "Updating to newer version");
+            MutableByteSpan updateToken(requestorCore->mUpdateTokenBuffer);
+            CopySpanToMutableSpan(update.updateToken, updateToken);
+            requestorCore->mUpdateVersion = update.softwareVersion;
```

Comment:
FYI, I recently made a change and this is now been renamed
```suggestion
            requestorCore->mTargetVersion = update.softwareVersion;
```

Reply:
Done.